tumblr/tumblr.php

```diff
@@ -247,11 +247,13 @@ function tumblr_settings(App $a, &$s)
 
 	$s .= '<div id="tumblr-enable-wrapper">';
 	$s .= '<label id="tumblr-enable-label" for="tumblr-checkbox">' . L10n::t('Enable Tumblr Post Addon') . '</label>';
+	$s .= '<input type="hidden" name="tumblr" value="0"/>';
```

Comment:
This confuses me. One line below we have got the input for exactly this name. I don't think that this will address the reported issue.

Reply:
This is a classic HTML feature: when a checkbox isn't checked, the associated field isn't sent at all in the POST request, which prevents the server to know whether the checkbox was unchecked (requiring data change) or disabled (preventing any data change).

This is a way of having the field always set in the POST request: if the checkbox is unchecked, the form falls back to the value of the hidden field with the same name. If you look at `field_checkbox.tpl`, this is how we handle all checkboxes in Friendica. This addon just doesn't use the templating engine.